tests/test_agent.py

```diff
@@ -1167,6 +1167,24 @@ def call_tool(messages: list[ModelMessage], info: AgentInfo) -> ModelResponse:
     )
 
 
+def test_output_type_text_output_function_with_deps():
```

Comment:
I'm pretty sure we've already tested somewhere that it _works_, this PR is just for fixing the type checking, so let's remove the new tests and add some cases to `typed_agent.py` instead.

Reply:
I think we can remove this test